experimental/builder/include/ck_tile/builder/reflect/instance_traits_util.hpp

```
@@ -0,0 +1,196 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2025, Advanced Micro Devices, Inc. All rights reserved.
+
+// Utility functions and helpers for instance_traits.hpp
+// Contains helper functions to convert types, enums, and sequences to string representations
+
+#pragma once
+
+#include <array>
+#include <string>
+#include <sstream>
+#include <type_traits>
+#include <ck/utility/data_type.hpp>
+#include <ck/utility/sequence.hpp>
+#include <ck/utility/blkgemmpipe_scheduler.hpp>
+#include <ck/tensor_operation/gpu/device/tensor_layout.hpp>
+#include <ck/tensor_operation/gpu/element/element_wise_operation.hpp>
+#include <ck/tensor_operation/gpu/device/convolution_forward_specialization.hpp>
+#include <ck/tensor_operation/gpu/device/gemm_specialization.hpp>
+
+namespace ck_tile::reflect::detail {
+
+// Metaprogramming helper to convert ck::Sequence to constexpr std::array
+template <typename Seq>
+struct SequenceToArray;
+
+template <ck::index_t... Is>
+struct SequenceToArray<ck::Sequence<Is...>>
+{
+    static constexpr std::array<int, sizeof...(Is)> value = {static_cast<int>(Is)...};
+};
+
+// Convert data types to string names
+template <typename T>
+inline std::string type_name()
+{
+    if constexpr(std::is_same_v<T, ck::half_t>)
+        return "fp16";
+    else if constexpr(std::is_same_v<T, float>)
+        return "fp32";
+    else if constexpr(std::is_same_v<T, double>)
+        return "fp64";
+    else if constexpr(std::is_same_v<T, int8_t>)
+        return "s8";
+    else if constexpr(std::is_same_v<T, int32_t>)
+        return "s32";
+    else if constexpr(std::is_same_v<T, ck::bhalf_t>)
+        return "bf16";
+    else if constexpr(std::is_same_v<T, ck::f8_t>)
+        return "fp8";
+    else if constexpr(std::is_same_v<T, ck::bf8_t>)
+        return "bf8";
+    else
+        return "unknown_type";
```

Rather than silently returning an invalid value, I suggest to make this function `consteval` and throw an error. When a developer adds a new type without patching this function, it would cause the compiler to emit an error (throwing is not allowed in consteval functions), reminding the programmer that they should add the type here. Example: https://godbolt.org/z/ee15KTjEz. (Also goes for the other functions in this file.)

---

That's so cool, thanks! I was wondering if there was a better way to implement this. I'm still getting used to C++20 idioms.